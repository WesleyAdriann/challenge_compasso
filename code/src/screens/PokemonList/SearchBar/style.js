import styled, { css } from 'styled-components/native'

import { COLORS, IMAGES } from '../../../style'

export const SearchBarWrapper = styled.View`
  flex-direction: row;
  height: 48px;
  margin: 4px 0;
  padding: 4px 8px;
`

const defaultStyle = css`
  border-color: ${COLORS.primaryBlack};
  border-radius: 200px;
  border-width: 1px;
  elevation: 3;
`

export const SearchInput = styled.TextInput`
  ${defaultStyle}
  background-color: white;
  flex-grow: 1;
  padding: 8px 12px;
`

export const SearchButton = styled.TouchableOpacity`
  ${defaultStyle}
  align-items: center;
  background-color: ${COLORS.primaryRed};
  justify-content: center;
  margin-left: 8px;
  width: 40px;
`

SearchButton.Icon = styled.Image.attrs(({ wasSearched }) => ({
  source: wasSearched ? IMAGES.icons.cancelIcon : IMAGES.icons.searchIcon
}))`
  height: 16px;
  width: 16px;
`
