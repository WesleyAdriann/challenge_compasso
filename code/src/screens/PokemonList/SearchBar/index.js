import React from 'react'
import PropTypes from 'prop-types'

import { IMAGES } from '../../../style'

import RoundButton from '../../../components/RoundButton'

import { SearchBarWrapper, SearchInput } from './style'

const SearchBar = ({ handleInputChange, searchValue, handleSearchPokemon, wasSearched }) => (
  <SearchBarWrapper>
    <SearchInput
      value={searchValue}
      onChangeText={handleInputChange('searchInput')}
      placeholder='Pokemon name or number'
    />
    <RoundButton onPress={handleSearchPokemon}>
      <RoundButton.Icon source={wasSearched ? IMAGES.icons.cancelIcon : IMAGES.icons.searchIcon} />
    </RoundButton>
  </SearchBarWrapper>
)

SearchBar.propTypes = {
  handleInputChange: PropTypes.func.isRequired,
  searchValue: PropTypes.string,
  handleSearchPokemon: PropTypes.func.isRequired,
  wasSearched: PropTypes.bool
}

SearchBar.defaultProps = {
  searchValue: '',
  wasSearched: false
}

export default SearchBar
