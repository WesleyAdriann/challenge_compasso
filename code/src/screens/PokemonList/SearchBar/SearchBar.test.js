import React from 'react'
import renderer from 'react-test-renderer'

import SearchBar from './index'

describe('SearchBar Component', () => {
  test('renders correctly', () => {
    const props = {
      handleInputChange: jest.fn(),
      handleSearchPokemon: jest.fn()
    }

    const tree = renderer.create(<SearchBar {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
