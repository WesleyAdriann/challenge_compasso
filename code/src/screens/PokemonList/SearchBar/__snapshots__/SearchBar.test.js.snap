// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "height": 48,
        "marginBottom": 4,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 4,
        "paddingBottom": 4,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 4,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    placeholder="Pokemon name or number"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#050505",
          "borderRadius": 200,
          "borderWidth": 1,
          "elevation": 3,
          "flexGrow": 1,
          "paddingBottom": 8,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 8,
        },
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D50000",
        "borderColor": "#050505",
        "borderRadius": 200,
        "borderWidth": 1,
        "elevation": 3,
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 0,
        "marginLeft": 8,
        "opacity": 1,
        "width": 40,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/icons/search.png",
        }
      }
      style={
        Array [
          Object {
            "height": 16,
            "resizeMode": "cover",
            "width": 16,
          },
        ]
      }
    />
  </View>
</View>
`;
